lib/facter/salts.rb

```diff
@@ -1,23 +1,24 @@
-Facter.add('salts') do
-  confine :kernel => "Linux"
-  confine { File.exist? '/etc/shadow' }
-  confine :facterversion do |version|
-    Gem::Version.new(version) >= Gem::Version.new('2.0.0')
-  end
-  # read
-  shadow = Facter::Util::Resolution.exec('cat /etc/shadow')
```

Comment:
Could you move these lines into the `setcode do ... end` block? `confine :kernel => "Linux"` should ensure that the fact is not evaluated on Windows.